### DISCUSSION TITLE
R CMD warning about vignette

### DISCUSSION THREAD

**Discussion by SudoWodo** [SudoWodo/StochOptim#15] [Q&A]
When I run `devtools::check()` on my Linux machine I get no errors, notes or warnings but when the same code gets run on Windows it gives me warnings. 

```

> checking whether package 'stochoptim' can be installed ... WARNING
  See below...

> checking re-building of vignette outputs ... WARNING
  Error(s) in re-building vignettes:
    ...
  --- re-building 'example.Rmd' using rmarkdown
  Quitting from lines 37-51 (example.Rmd) 
  Error: processing vignette 'example.Rmd' failed with diagnostics:
  object 'method' not found
  --- failed re-building 'example.Rmd'
  
  SUMMARY: processing the following file failed:
    'example.Rmd'
  
  Error: Vignette re-building failed.
  Execution halted

0 errors v | 2 warnings x | 0 notes v
```

I got warnings resolved on Linux by installing `qpdf`. But in Windows it suggests that a method is missing but I it's not.
Please help.

**Comment by nashjc** [comment]
I wonder if the name "example" is the problem. Perhaps try changing name of vignette to Xample.Rmd.

Looks like one of the many inconveniences of cross platform computing. 

Another option is to build with the option to NOT build vignettes. I think --no-build-vignettes. I'm not sure how to set that
in Rstudio, but have used it in command line.

**Reply by SudoWodo** [reply]
I'll try that. Thank you

**Comment by SudoWodo** [comment]
## Rstudio build options
I tried in Rstudio by going in Build > Configure build tools than tried the following in Check package text field: 
`--ignore-vignettes` `--no-vignettes`

That didn't help.

## Devtools
While checking the package with `devtools::check()`  simply passing the vignettes = FALSE did the trick. `devtools::check(vignettes = FALSE)`

**Reply by SudoWodo** [reply]
Now it is throwing just one warning which is :- 
```{r}
Found the following significant warnings:
     Warning: package 'R6' was built under R version 4.1.3
```

**Reply by nashjc** [reply]
the devtools::check() may be the only way to do this within Rstudio. 

I'm not sure what to say about R6 version issue. R6 is from Winston Chang who is an Rstudio employee. Possible Rstudio wants
us to use a version that is more recent than CRAN has. 

At some point it may be worth thinking of whether R6 is strictly needed for our use. The fewer dependencies the better in my experience.

**Reply by SudoWodo** [reply]
With R6 we are reusing a lot of code! So for now I think I'll keep it.

I update R to `version 4.2.0 (2022-04-22 ucrt) -- "Vigorous Calisthenics"` and now everything works perfectly, no errors, warnings or notes.
